src/etc/gdb_providers.py

```diff
@@ -0,0 +1,406 @@
+from sys import version_info
+
+import gdb
+from gdb import lookup_type
+
+if version_info[0] >= 3:
+    xrange = range
+
+ZERO_FIELD = "__0"
+FIRST_FIELD = "__1"
+
+
+def unwrap_unique_or_non_null(unique_or_nonnull):
+    # BACKCOMPAT: rust 1.32
+    # https://github.com/rust-lang/rust/commit/7a0911528058e87d22ea305695f4047572c5e067
+    ptr = unique_or_nonnull["pointer"]
+    return ptr if ptr.type.code == gdb.TYPE_CODE_PTR else ptr[ZERO_FIELD]
+
+
+class StructProvider:
+    def __init__(self, valobj):
+        self.valobj = valobj
+        self.fields = self.valobj.type.fields()
+
+    def to_string(self):
+        return self.valobj.type.name
+
+    def children(self):
+        for field in self.fields:
+            yield field.name, self.valobj[field.name]
+
+
+class TupleProvider:
```

Comment:
I don't understand the benefit of this, either.

Reply:
It replaces `__0` with `0` and so on. Should I get rid of this provider and leave the `__` prefix?